src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo.cs

```diff
@@ -212,76 +191,46 @@ public SymbolTreeInfo WithChecksum(Checksum checksum)
             bool ignoreCase,
             CancellationToken cancellationToken)
         {
-            var comparer = GetComparer(ignoreCase);
-            IAssemblySymbol? assemblySymbol = null;
-
             using var results = TemporaryArray<ISymbol>.Empty;
-            foreach (var node in FindNodeIndices(name, comparer))
+
+            var (startIndexInclusive, endIndexExclusive) = FindCaseInsensitiveNodeIndices(_nodes, name);
```

Comment:
moved away from an allocation `IEnumerable<int>` and just returned a simple range.